src/main/io/rcdevice.c

```diff
@@ -0,0 +1,1013 @@
+        }
+        rcsplitSendCtrlCommand(device, operation + 1);
+    } else {
+        runcamDeviceSendPacket(device, RCDEVICE_PROTOCOL_COMMAND_CAMERA_CONTROL, &operation, sizeof(operation));
+    }
+
+    return true;
+}
+
+// every time start to control the OSD menu of camera, must call this method to
+// camera
+bool runcamDeviceOpen5KeyOSDCableConnection(runcamDevice_t *device)
+{
+    uint8_t actionID = 0xFF;
+    uint8_t code = 0xFF;
+    bool r = runcamDeviceSend5KeyOSDCableConnectionEvent(device, RCDEVICE_PROTOCOL_5KEY_CONNECTION_OPEN, &actionID, &code);
+    return r && (code == 1) && (actionID == RCDEVICE_PROTOCOL_5KEY_CONNECTION_OPEN);
+}
+
+// when the control was stop, must call this method to the camera to disconnect
+// with camera.
+bool runcamDeviceClose5KeyOSDCableConnection(runcamDevice_t *device)
+{
+    uint8_t actionID = 0xFF;
+    uint8_t code = 0xFF;
+    bool r = runcamDeviceSend5KeyOSDCableConnectionEvent(device, RCDEVICE_PROTOCOL_5KEY_CONNECTION_CLOSE, &actionID, &code);
+
+    return r && (code == 1) && (actionID == RCDEVICE_PROTOCOL_5KEY_CONNECTION_CLOSE);
+}
+
+// simulate button press event of 5 key osd cable with special button
+bool runcamDeviceSimulate5KeyOSDCableButtonPress(runcamDevice_t *device, uint8_t operation)
+{
+    if (operation == RCDEVICE_PROTOCOL_5KEY_SIMULATION_NONE) {
+        return false;
+    }
+
+    if (runcamDeviceSendRequestAndWaitingResp(device, RCDEVICE_PROTOCOL_COMMAND_5KEY_SIMULATION_PRESS, &operation, sizeof(uint8_t), NULL, NULL)) {
+        return true;
+    }
+
+    return false;
+}
+
+// simulate button release event of 5 key osd cable
+bool runcamDeviceSimulate5KeyOSDCableButtonRelease(runcamDevice_t *device)
+{
+    return runcamDeviceSendRequestAndWaitingResp(device, RCDEVICE_PROTOCOL_COMMAND_5KEY_SIMULATION_RELEASE, NULL, 0, NULL, NULL);
+}
+
+// fill a region with same char on screen, this is used to DisplayPort feature
+// support
+void runcamDeviceDispFillRegion(runcamDevice_t *device, uint8_t x, uint8_t y, uint8_t width, uint8_t height, uint8_t c)
+{
+    uint8_t paramsBuf[5];
+
+    // fill parameters buf
+    paramsBuf[0] = x;
+    paramsBuf[1] = y;
+    paramsBuf[2] = width;
+    paramsBuf[3] = height;
+    paramsBuf[4] = c;
+
+    runcamDeviceSendPacket(device, RCDEVICE_PROTOCOL_COMMAND_DISP_FILL_REGION, paramsBuf, sizeof(paramsBuf));
+}
+
+// draw a single char on special position on screen, this is used to DisplayPort
+// feature support
+void runcamDeviceDispWriteChar(runcamDevice_t *device, uint8_t x, uint8_t y, uint8_t c)
+{
+    uint8_t paramsBuf[3];
+
+    // fill parameters buf
+    paramsBuf[0] = x;
+    paramsBuf[1] = y;
+    paramsBuf[2] = c;
+
+    runcamDeviceSendPacket(device, RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_CHAR, paramsBuf, sizeof(paramsBuf));
+}
+
+// draw a string on special position on screen, this is used to DisplayPort
+// feature support
+void runcamDeviceDispWriteHortString(runcamDevice_t *device, uint8_t x, uint8_t y, const char *text)
```

Comment:
No unnecessary abbreviations. This should be `runcamDeviceDispWriteHorizontalString`